src/query/models/tags.go

```diff
@@ -93,37 +88,118 @@ func (t Tags) IDLen() int {
 	return idLen
 }
 
-// IDWithExcludes returns a string representation of the tags excluding some tag keys.
-func (t Tags) IDWithExcludes(excludeKeys ...[]byte) uint64 {
-	b := make([]byte, 0, t.Len())
-	for _, tag := range t.Tags {
-		// Always exclude the metric name by default
-		if bytes.Equal(tag.Name, t.Opts.MetricName()) {
-			continue
+func (t Tags) quotedID() []byte {
+	needEscaping, length := t.escapingAndLength()
+	if needEscaping == nil {
+		return t.quoteIDSimple(length)
+	}
+
+	// TODO: pool these bytes.
+	id := make([]byte, length)
+	idx := 0
+	for i, tag := range t.Tags {
```

Comment:
A potential readability refactor for your consideration: could we build up this function in terms of a method to write out a single tag to a `[]byte`? Something like:

```
length := 0
for _, t :=` range t.Tags {
    sliceLen += t.SerializedLength()  // this would take into account NeedsEscaping, etc
}

id := make([]byte, length)
for _, t := range t.Tags {
    t.WriteTo(id)
}
```